services/src/contexts/migrations/migration_0016_merge_providers.sql

```diff
@@ -0,0 +1,35 @@
+ALTER TYPE "DataProviderDefinition" ADD ATTRIBUTE
+sentinel_s2_l2_a_cogs_provider_definition
+"SentinelS2L2ACogsProviderDefinition";
+
+ALTER TYPE "DataProviderDefinition" ADD ATTRIBUTE
+copernicus_dataspace_provider_definition
+"CopernicusDataspaceDataProviderDefinition";
+
+INSERT INTO layer_providers (
+    id,
+    type_name,
+    name,
+    definition
+) SELECT
+    pro.id,
+    pro.type_name,
+    pro.name,
+    (
+        NULL,
+        NULL,
+        NULL,
+        NULL,
+        NULL,
+        NULL,
+        NULL,
+        NULL,
+        NULL, -- noqa: PRS
+        (pro.definition).sentinel_s2_l2_a_cogs_provider_definition,
+        NULL
+    )::"DataProviderDefinition"
+FROM pro_layer_providers AS pro
+WHERE pro.type_name = 'SentinelS2L2ACogsProviderDefinition';
```

Comment:
what about the CopernicusDataspaceDataProviderDefinition?

Reply:
[ed8a449](https://github.com/geo-engine/geoengine/pull/1009/commits/ed8a449d12e59f370bafe5dcbc4313dfcde06cf6)